### PR TITLE
boards/nucleo-f410: initial support

### PR DESCRIPTION
**Pull request by aabadie**
We have plenty of nucleo boards. This PR ports nucleo64 f410 to RIOT.

Contrary to other STM32f4 already available in RIOT, the STM32f410rb RCC peripheral doesn't use the same register for RNG. I made a small change to make this configurable in the periph_cpu.h file.

### CODE REVIEW

File: cpu/stm32f4/periph/timer.c [hunk 1]
```diff
@@ -273,12 +273,16 @@ void timer_irq_disable(tim_t dev)
 
 void TIMER_0_ISR(void)
 {
+#if TIMER_0_EN
```
Comment (haukepetersen) [inline cpu/stm32f4/periph/timer.c]: Please change to
```c
#ifdef TIMER_0_ISR
void TIMER_0_ISR(void) 
{
    irq_handler(TIMER_0, TIMER_0_DEV);
}
#endif

**Comment by haukepetersen** [comment]
Appart from the minor remark I think everything is in order here..

**Comment by PeterKietzmann** [comment]
Looks fine though I'd like to check hwrng on stm32f4 before merge. Already started Murdock.

**Comment by PeterKietzmann** [comment]
@aabadie relic unittests overflow ROM on that device. Please blacklist.

**Comment by aabadie** [comment]
@PeterKietzmann, updated the tests. Let's see what Murdock says.
I tested the port on hardware : it works but requires openocd 0.10...

**Comment by PeterKietzmann** [comment]
@aabadie let's wait for #4780.

**Comment by aabadie** [comment]
@PeterKietzmann np the delay due to #4780 was expected.

**Comment by astralien3000** [comment]
No PWM is configured, I guess it is normal.

I tested : 
 - examples/hello_world
 - examples/default (saul  LED+BUTTON)
 - tests/periph_uart
 - tests/periph_timer
 - tests/periph_hwrng
 - tests/xtimer_msg

Everything is OK.

**Comment by aabadie** [comment]
> No PWM is configured, I guess it is normal.

We could configure some on the ST Morpho connectors but none are available on the arduino connectors.

**Comment by kYc0o** [comment]
Is this a nucleo32, 64, 144? should be in the folder's name?

**Comment by aabadie** [comment]
> Is this a nucleo32, 64, 144? should be in the folder's name?

nucleo 64. This is the implicit convention I choosed (see [comment](https://github.com/RIOT-OS/RIOT/pull/6417#issue-201692829)):
* nucleo32 -> 32 pins version (Arduino nano compatible)
* nucleo -> 64 pins version (Arduino connectors + ST morpho connectors)
* nucleo144 -> 144 pins version

**Comment by kaspar030** [comment]
Why did Jenkins build this correctly?

**Comment by aabadie** [comment]
> Why did Jenkins build this correctly?

Maybe it didn't trigger a new build in case of a conflict.
Anyway, I rebased. Let's see if it's green again now.

**Comment by kYc0o** [comment]
Ups, I think this broke #6740. Can you re-sort alphabetically the `BOARDS_INSUFFICIENT_MEMORY` in tests/unittests and relic?

**Comment by aabadie** [comment]
@kYc0o good catch. For unittest, it's ok not for relic.

**Comment by aabadie** [comment]
@kYc0o, I opened #6746

**Pull request merged**